includes/Database_Upgrader.php

```diff
@@ -0,0 +1,100 @@
+<?php
+/**
+ * Class Database_Upgrader
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories;
+
+/**
+ * Class Database_Upgrader
+ *
+ * @package Google\Web_Stories
+ */
+class Database_Upgrader {
+
+	/**
+	 * The slug of database option.
+	 *
+	 * @var string
+	 */
+	const OPTION = 'web_stories_db_version';
+
+	/**
+	 * The slug of database option.
+	 *
+	 * @var string
+	 */
+	const PREVIOUS_OPTION = 'web_stories_previous_db_version';
+
+	/**
+	 * Hooked into admin_init and walks through an array of upgrade methods.
+	 *
+	 * @return void
+	 */
+	public function init() {
+		$version  = get_option( self::OPTION, '0.0.0' );
+		$routines = [
+			'1.0.0' => 'upgrade_1',
+		];
+
+		array_walk( $routines, [ $this, 'run_upgrade_routine' ], $version );
+		$this->finish_up( $version );
+	}
+
+	/**
+	 * Runs the upgrade routine.
+	 *
+	 * @param string $routine         The method to call.
+	 * @param string $version         The new version.
+	 * @param string $current_version The current set version.
+	 *
+	 * @return void
+	 */
+	protected function run_upgrade_routine( $routine, $version, $current_version ) {
+		if ( version_compare( $current_version, $version, '<' ) ) {
+			$this->$routine( $current_version );
+		}
+	}
+
+	/**
+	 * First database migration.
+	 *
+	 * @return void
+	 */
+	protected function upgrade_1() {
+		// Do nothing.
+	}
+
+	/**
+	 * Runs the needed cleanup after an update, setting the DB version to latest version, flushing caches etc.
+	 *
+	 * @param string $previous_version The previous version.
+	 *
+	 * @return void
+	 */
+	protected function finish_up( $previous_version ) {
+		update_option( self::PREVIOUS_OPTION, $previous_version );
+		update_option( self::OPTION, constant( 'WEBSTORIES_DB_VERSION' ) );
```

Comment:
```suggestion
		update_option( self::OPTION, WEBSTORIES_DB_VERSION );
```

Reply:
Why is this change being requested?

Reply:
Using `constant()` is superfluous here. Citing the [docs](https://www.php.net/manual/en/function.constant.php):

> `constant()` is useful if you need to retrieve the value of a constant, but do not know its name. I.e. it is stored in a variable or returned by a function.

That's not the case. Let's not add unnecessary code 🙂

Reply:
I always use `constant()` is if the value of that constant is not set for some reason, it fails gracefully. If we don't use `constant()` when this function call should be wrapped in `defined`.

Reply:
But `WEBSTORIES_DB_VERSION` is always guaranteed to be available as it's defined in the main plugin file.

Since we don't currently use `constant()` anywhere at the moment, and enforcing that is pretty much a coding standards decision, let's omit this here and discuss this separately.

Reply:
I removed the constant call and phpstan failed. I really think we should keep this call in.

Reply:
That's because the constant needs to be added to `tests/phpstan/bootstrap.php`, which is how phpstan works.

Reply:
> That's because the constant needs to be added to `tests/phpstan/bootstrap.php`, which is how phpstan works.

It is for reasons like this why I used `constant` in the first place.

Reply:
I'd say this particular example is an argument for the opposite 🙂 

In any case, I am not totally opposed to using `constant()`, but please let's discuss this separately instead of silently introducing using it in this PR.

If you can the usage here & amend the `boostrap.php` file, then this is good to go.

Reply:
Fixed in https://github.com/google/web-stories-wp/pull/1751/commits/16e7690968881e6e01286d92a669ff0497ed715c